src/NetEscapades.EnumGenerators/EquatableArray.cs

```diff
@@ -8,14 +8,20 @@ namespace NetEscapades.EnumGenerators;
 /// An immutable, equatable array. This is equivalent to <see cref="Array{T}"/> but with value equality support.
 /// </summary>
 /// <typeparam name="T">The type of values in the array.</typeparam>
-public readonly struct EquatableArray<T> : IEquatable<EquatableArray<T>>, IEnumerable<T>
+public readonly struct EquatableArray<T> : IEquatable<EquatableArray<T>>, IReadOnlyList<T>
     where T : IEquatable<T>
 {
     /// <summary>
     /// The underlying <typeparamref name="T"/> array.
     /// </summary>
     private readonly T[]? _array;
 
+    /// <summary>
+    /// Get value at <paramref name="index"/>
+    /// </summary>
+    /// <param name="index">The index of the value to get</param>
+    public T this[int index] => _array[index];
+
```

Comment:
```suggestion
```

Reply:
That's the one addition for `IReadOnlyList<T>`, the members for `IReadOnlyCollection<T>` were already there. In a way I also expect an indexed-Property for a type with `Array` in the name. I would like to keep it, and it should neither raise the complexity nor cause other trouble